### ISSUE TITLE
local tunnel service stoped providing a link

### ISSUE THREAD

**Issue by mcorcias** [localtunnel/localtunnel#664]
I haven't received a link for several hours, Is There A Problem?

**Comment by aleksandr-b** [comment]
Same here...

**Comment by ReactIntakeNative** [comment]
Same here, have been trying for an hour

**Comment by Aritra-Khanra** [comment]
has it worked for any of you yet? I am not receiving any links.

**Comment by Tuan-TIT** [comment]
Same here. yesterday, the issue occurred at about 11:00 PM (UTC +7)

**Comment by lacherogwu** [comment]
👀

**Comment by Pamigo79** [comment]
Same here. https://localtunnel.me is down as well. Yesterday I was only able to get it working for an hour or so as well.

**Comment by TheBoroer** [comment]
Hey all, the server is ultra slow right now...
Seems like some people are hardcore spamming the server with tunnel requests causing all 8 cores to max out. 
I'll need some time to figure out how to mitigate/fix this

**Comment by MarkNCI** [comment]
Same for me, running localtunnel in Python and no link provided. Can we spam the spammers :P

**Comment by azzamalzid** [comment]
> Hey all, the server is ultra slow right now... Seems like some people are hardcore spamming the server with tunnel requests causing all 8 cores to max out. I'll need some time to figure out how to mitigate/fix this

What we can do to help you? we are with you

**Comment by Aritra-Khanra** [comment]
any updates, is it working now?

**Comment by greatwolf** [comment]
yea, I'm getting the same problem just to add another data point.

**Comment by Parking-Master** [comment]
@TheBoroer I'd say the best thing to do is shut down the servers until the spammers go away. Leaving it on and letting the spammers continue slowing the server down isn't a good idea since it may take longer for localtunnel to come back up.

**Comment by greatwolf** [comment]
as much as I hate captchas, what about adding one to the initial start page when it ask you for the ip of the server or is it already too late in the process?

**Comment by ReactIntakeNative** [comment]
Still not working.

**Comment by zfareed** [comment]
same issue, tunnel is not starting from last 2-3 days.

**Comment by sk-tjdownes** [comment]
We've been using LT for about a year and it's been great. We have to test several webhooks in our environments, and other solutions are too costly for our organization's budget.

However, with the issue LT is experiencing lately, it hasn't been a dependable solution. 

My thought here is that LT should be extended to allow teams to configure their own tunneling servers. Hosting your own tunneling server on AWS would be pretty inexpensive, and remove the burden on @TheBoroer to host it. I assume the bulk of the work is essentially documenting how to build the tunneling server, or maybe just providing a Docker image?

What's the feasibility of this?

**Comment by greatwolf** [comment]
Do you mean something like this? https://github.com/localtunnel/server?tab=readme-ov-file#deploy

**Comment by shivammalviya71234** [comment]
> Do you mean something like this? https://github.com/localtunnel/server?tab=readme-ov-file#deploy

It isn't very clear to me on how to make it work. Do you have link to any good article describing that?

**Comment by sk-tjdownes** [comment]
> Do you mean something like this? https://github.com/localtunnel/server?tab=readme-ov-file#deploy

yep! RTFM, I guess ;)

**Comment by greatwolf** [comment]
> It isn't very clear to me on how to make it work. Do you have link to any good article describing that?


Not really, have to do some trial and error to figure it out. Besides the docker command there, you'll need a dns service that supports wildcard so that subdomains generated on the fly can resolve to the correct server ip. Cloudns seems to support that, or you can use something like nip.io.

The other thing you need to do is add `--domain` option to the docker command and point it to the domain name the server will be using. For example, say your localtunnel server is `111.222.33.44`, `111.222.33.44.nip.io` would resolve to that ip, the docker command would be:

```
docker run -d \
    --restart always \
    --name localtunnel \
    --net host \
    defunctzombie/localtunnel-server:latest --port 3000 --domain 111.222.33.44.nip.io
```

And the last thing you need to do is make sure `lt` command on the client side uses the `--host` option to point to your server. eg. `lt --host http://111.222.33.44.nip.io:3000 --port 9000`

Without it, it'll default to localtunnel.me

**Comment by TheBoroer** [comment]
Hey all! Sorry I've been MIA. I'll read all the missed messages and reply where need be 🫡 

I've been trying to figure out the cause of the high CPU load on the server, which is the cause of all these issues. Traefik was hogging all the CPU cores for some reason...i'm guessing it's SSL handshake related or maybe a silly misconfiguration on my end. Anyways, I've switched back to using nginx again and the CPU load is way lower so we should be ok now!

I still see some random 502 errors sometimes on the tunnel reminder page so I'll investigate that (not sure if it happens to tunneled content too though)

I appreciate all the patience and sorry for the downtime everyone 🙏

**Comment by Parking-Master** [comment]
@TheBoroer no worries! We're just glad you found a solution. 😀

**Comment by sk-tjdownes** [comment]
@TheBoroer thank you. We all know open source is a thankless job, filled with constant asks and very little reward. We appreciate your dedication to keeping lt running!

**Comment by Toriong** [comment]
It seems to be down again. Or maybe it's just me? Anybody else dealing with this issue?

**Comment by LWJerri** [comment]
> It seems to be down again. Or maybe it's just me? Anybody else dealing with this issue?

Nah, it's also happen for me.

**Comment by greatwolf** [comment]
yup, I'm seeing the same.

**Comment by shivammalviya71234** [comment]
It's working for me, but yesterday I was facing some issues.

**Comment by hmidmrii** [comment]
Any updates here? I'm still facing the issue

**Comment by Parking-Master** [comment]
@hmidmrii Don't worry, it happens off and on like this. The only thing you can do is wait for it to start working again. It shouldn't be long, just about an hour or so.

**Comment by TheBoroer** [comment]
> Any updates here? I'm still facing the issue

Hey, everything seems to be working atm. Please note; it might sometimes be slow to give you a link but it will give you one eventually. 

I'm looking into how i can tweak the localtunnel server so that it can run over several instances instead of just 1 (which sometimes gets overloaded with spam requests)

**Comment by adearriba** [comment]
Hi @TheBoroer, do you have plans to update the server repo? 

I deployed it in my own server but for some reason there is a strange connection bug I've been trying to fix for 2 weeks now without luck. After some time, the remote socket puts in read-only mode and the local socket ends. However, the remote server is unable to end even if I force it. Not sure why. Maybe your last version has fixed this somehow. When using your server it all works without issues.

I've tried multiple things. Even a full refactor. Nothing. It's frustrating.

**Comment by TheBoroer** [comment]
> Hi [@TheBoroer](https://github.com/TheBoroer), do you have plans to update the server repo?

Sorry for the late reply, feel free to checkout my fork of the server over at https://github.com/TheBoroer/localtunnel-server (my fork is also auto-built into a docker image which is available at: https://hub.docker.com/r/boro/localtunnel-server). I'm not sure if any of the fixes there will address the issue you're seeing. I'll checkout #673 you opened and see if the issue exists in my fork.

I haven't merged anything into the official repos because I want to setup proper testing before merging stuff that could break for others.

PS. I've also setup a service status page (see #697) for better monitoring.
